pymc_marketing/clv/models/beta_geo_beta_binom.py

```diff
@@ -0,0 +1,364 @@
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import pymc as pm
+import xarray
+from numpy import exp
+from pytensor.tensor import TensorVariable
+from scipy.special import betaln
+from scipy.special import gamma as gammaf
+from scipy.special import gammaln
+
+from pymc_marketing.clv.distributions import BetaGeoBetaBinom
+from pymc_marketing.clv.models.basic import CLVModel
+from pymc_marketing.clv.utils import to_xarray
+
+
+# TODO: Proofread docstrings
+class BetaGeoBetaBinomModel(CLVModel):
+    r"""Beta-Geometric Beta Binomial Distribution (BG/BB) model for discrete, non-contractual transactions.
+
+    In the BG/BB model, the frequency of customer purchases is modelled as the time
+    of each purchase has an instantaneous probability of occurrence (hazard) and, at
+    every purchase, a probability of "dropout", i.e. no longer being a customer.
+
+    Customer-specific data needed for statistical inference include 1) the total
+    number of purchases (:math:`x`) and 2) the time of the last, i.e. xth, purchase. The
+    omission of purchase times :math:`t_1, ..., t_x` is due to a telescoping sum in the
+    exponential function of the joint likelihood; see Section 4.1 of [1] for more
+    details.
+
+    Parameters
+    ----------
+    customer_id: array_like
+        Customer labels. Must not repeat.
+    frequency: array_like
+        The number of purchases of customers.
+    recency: array_like
+        The time of the last, i.e. xth, purchase.
```

Comment:
We can perhaps make a point that `recency` is discrete, i.e. `int`-like.

Reply:
Yeah that docstring still requires revision. Could be a good idea to add a validation check for this as well.